webapp-django/crashstats/crashstats/views.py

```diff
@@ -136,6 +140,68 @@ def build_data_object_for_crash_reports(response_items):
     return crash_reports
 
 
+def get_report_list_from_super_search(**kwargs):
+    '''Return results as expected by the report_list views, but loaded
+    from Super Search.
+
+    This function converts given parameters to match what Super Search expects.
+    It changes some values (like version and _sort parameters). It then
+    queries Super Search, and maps the field names in the results to what
+    the report/list/ views and templates expect.
+    '''
+    # Prepare mapping of names to in_database_names.
+    fields = SuperSearchFields().get()
+    old_names_map = {}
+    for field in fields.values():
+        if (
+            (field['is_exposed'] or field['is_returned']) and
+            field['name'] != field['in_database_name']
+        ):
+            old_names_map[field['in_database_name']] = field['name']
+
+    # Convert parameters to what Super Search accepts.
+    params = kwargs.copy()
+    if params.get('signature'):
+        params['signature'] = '=' + params['signature']
+
+    if params.get('end_date') or params.get('start_date'):
+        params['date'] = []
+        if 'start_date' in params:
+            start_date = params['start_date'].isoformat()
+            params['date'].append('>=' + start_date)
+            del params['start_date']
+        if 'end_date' in params:
+            end_date = params['end_date'].isoformat()
+            params['date'].append('<=' + end_date)
+            del params['end_date']
+
+    if params.get('version'):
+        versions = [
+            x.split(':')[-1] for x in params['version'] if x != 'ALL:ALL'
+        ]
+        params['version'] = versions
+
+    for key in ('process_type', 'hang_type'):
+        # For those keys, 'any' is not supported and should be an empty value.
+        if params.get(key) == 'any':
+            del params[key]
+
+    for param in params:
+        if param in old_names_map:
+            params[old_names_map[param]] = params[param]
+            params[param] = None
```

Comment:
why doesn't that do `del params[param]`? Perhaps I'm not understanding the code but basically this means that the key is not a known, SuperSearch field, right?
Perhaps what I'm saying is that this code deserves a human-friendly comment.